backend/src/test/scala/cromwell/backend/standard/callcaching/BlacklistCacheSpec.scala

```diff
@@ -1,22 +1,44 @@
 package cromwell.backend.standard.callcaching
 
 import cromwell.core.CacheConfig
-import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+import cromwell.services.CallCaching.CallCachingEntryId
+import org.scalatest.concurrent.Eventually
+import org.scalatest.{FlatSpec, Matchers}
 
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
-class BlacklistCacheSpec extends FlatSpec with BeforeAndAfterAll with Matchers {
-  "The blacklist cache" should "default, blacklist and expire" in {
+class BlacklistCacheSpec extends FlatSpec with Matchers with Eventually {
+  "The blacklist cache" should "default, blacklist, whitelist and expire" in {
+    val hit = CallCachingEntryId(3)
     val bucket = "foo"
-    val cacheConfig = CacheConfig(concurrency = 1, size = Integer.MAX_VALUE, ttl = 1 second)
-    val cache = BlacklistCache(cacheConfig)
-    cache.isBlacklisted(bucket) shouldBe false
+
+    val bucketCacheConfig = CacheConfig(concurrency = 1, size = Integer.MAX_VALUE, ttl = 1 second)
+    val hitCacheConfig = CacheConfig(concurrency = 1, size = Integer.MAX_VALUE, ttl = 1 second)
+    val cache = new RootWorkflowBlacklistCache(bucketCacheConfig = bucketCacheConfig, hitCacheConfig = hitCacheConfig)
+    cache.getBlacklistStatus(bucket) shouldBe UntestedCacheResult
+    cache.getBlacklistStatus(hit) shouldBe UntestedCacheResult
     cache.blacklist(bucket)
-    cache.isBlacklisted(bucket) shouldBe true
+    cache.blacklist(hit)
+    cache.getBlacklistStatus(bucket) shouldBe BadCacheResult
+    cache.getBlacklistStatus(hit) shouldBe BadCacheResult
+
+    implicit val patienceConfig = PatienceConfig(timeout = scaled(5.seconds), interval = scaled(1.second))
 
     // Test ttl
-    Thread.sleep(5000L)
-    cache.isBlacklisted(bucket) shouldBe false
+    eventually {
+      cache.getBlacklistStatus(bucket) shouldBe UntestedCacheResult
```

Comment:
TOL it just occurred to me that continuous polling might actually reset the liveness counter, so maybe this was better as it was before after all? 🤔 

If so, sorry for making you change it!

Reply:
Hmm actually this seems to work just fine, I guess TTL is based on last write and not last read.